heron/spi/src/java/com/twitter/heron/spi/utils/NetworkUtils.java

```diff
@@ -397,6 +397,15 @@ public static boolean checkHttpResponseCode(HttpURLConnection connection, int ex
     }
   }
 
+  public static Integer getHttpResponseCode(HttpURLConnection connection) {
+    try {
+      return connection.getResponseCode();
+    } catch (IOException ex) {
+      LOG.log(Level.SEVERE, "Failed to get response code");
+      return null;
```

Comment:
I'm really not as fan of returning null to represent exceptions (see rant in #1311). If we can't get a response code it's an exceptional case, so I think we should throw an exception.